modules/store/src/selector.ts

```diff
@@ -17,13 +17,33 @@ export type ComparatorFn = (a: any, b: any) => boolean;
 
 export type DefaultProjectorFn<T> = (...args: any[]) => T;
 
+type StrictMemoizedSelector<
+  State,
+  Slices extends unknown[],
+  Result
+> = MemoizedSelector<
+  State,
+  Result,
+  { strict: true } & ((...slices: Slices) => Result)
```

Comment:
`{ strict: true }` is purely type information which is used by `MemoizedSelector` to determine which projector function is used.

Reply:
Apparently this is known as an [opaque type](https://calebmer.com/tangents/opaque-types-in-typescript.html).